### ISSUE TITLE
[BUG] F-Droid can't build

### ISSUE THREAD

**Issue by licaon-kter** [DanXi-Dev/DanXi#333]
ref: https://gitlab.com/fdroid/fdroiddata/-/jobs/6342664164#L1663

/LE: fyi https://gitlab.com/fdroid/fdroiddata/-/commit/3a5585a2396a6208d1ce06bd08bcb808e16b8330

**Comment by w568w** [comment]
Thanks for your reminder. I think it is due to https://github.com/DanXi-Dev/DanXi/commit/dc5cac123f97a4a35e406694bc3742971e780f31, where all auto-generated files were removed from the repository. 

For some reason [the F-Droid metadata](https://gitlab.com/fdroid/fdroiddata/-/blob/3a5585a2396a6208d1ce06bd08bcb808e16b8330/metadata/io.github.danxi_dev.dan_xi.yml) does not execute `.flutter/bin/dart run build_runner build --delete-conflicting-outputs` to regenerate them. Adding this line before `.flutter/bin/flutter build apk --release` in `build` commands should fix this.

However, a better solution may be to use the new build script, [`build_release.dart`](https://github.com/DanXi-Dev/DanXi/blob/main/build_release.dart), which adds necessary command arguments to build the app with extra information (e.g. git hash). I will try to integrate that with `build` commands soon.

**Comment by licaon-kter** [comment]
@w568w well, as you see fit, we can add that of course but if it can be integrated maybe it's better

ping when ready to test

**Comment by w568w** [comment]
MR is merged. Close it.

**Comment by licaon-kter** [comment]
https://gitlab.com/fdroid/checkupdates-bot-fdroiddata/-/jobs/8500452450

is this fixed by the last commit?

**Comment by w568w** [comment]
> is this fixed by the last commit?

Yes, it has been fixed by 144fd82508c7a3c17a44f19997503caa80fa206f. Thanks for your notification. I will merge the fix into the `foss-build` branch very soon. 

By the way, to rerun the F-Droid build on that commit, should I submit a merge request, or should I use `git push --tags -f` to move the tag `foss-v1.4.5` to the new commit?

**Comment by licaon-kter** [comment]
I'm testing in https://gitlab.com/fdroid/fdroiddata/-/merge_requests/17038 right now

**Comment by w568w** [comment]
F-Droid build error: `DEBUG: Problem: found class 'repeackage/com/google/android/gms/ads/identifier/internal/IAdvertisingIdService'`

**Comment by w568w** [comment]
> this gets triggered by matching `com/google/android/gms` (per https://gitlab.com/fdroid/fdroid-suss/-/blob/master/suss/com.google.android.gms.yml#L25)
> 
> then again, where does it come from?

It turns out that this package introduced the interface `com.google.android.gms.ads.identifier.internal.IAdvertisingIdService`:

https://github.com/DanXi-Dev/DanXi/blob/271a65a4f56fa954265ef102a71f990480efb620/pubspec.yaml#L97

which depends on [Android_CN_OAID](https://github.com/gzu-liyujiang/Android_CN_OAID), an unified SDK to access private ROM-specific APIs that provide device IDs.

In this project, we only relied on this library for device registration of message push service. However, the messaging SDK has been removed from the FOSS branch, so this package is unnecessary. 

I will remove this dependency soon. When I get done, I will ping you at [the downstream side](https://gitlab.com/fdroid/fdroiddata/-/merge_requests/17038).
